### PR TITLE
[Router] Add <Connect> 

### PR DESCRIPTION
**Pull request by damassi**
(**Note:** Broke this PR off of https://github.com/artsy/reaction/pull/1118 since it diverged.)

Adds a new component `<Connect>` which aliases `<Subscribe to={State}>` for a clearer expression of purpose when needing to extract values. This follows language similar to Redux:

```jsx
import { Connect } from 'Router'

<Connect to={AppState}>
  {({ mediator }) => 
    <Hello onClick={() => mediator.trigger()} />
  }}
</Connect>
``` 

Updates `Subscribe` imports to reference an export alias from Router for internal consistency:

```
import { AppState, Connect, Subscribe, etc } from 'Router'
```

@alloy, @zephraph and @mzikherman, q: 

I'm thinking that `/Router` is probably the wrong name for this directory since its more of a light-weight framework at this point. Thoughts?

### CODE REVIEW

**Comment by artsyit** [comment]
Deploy preview for *artsy-reaction* ready!

Built with commit 58fb7088cbc7590c3c98f485e38c8ce5c6fac232

https://deploy-preview-1119--artsy-reaction.netlify.com

File: src/Router/Connect.tsx [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    | Array<ContainerType<any> | Container<any>>
+  children(...instances: any[]): React.ReactNode
+}
+
+export const Connect: SFC<ConnectProps> = ({ to, children }) => {
+  const _to = isArray(to) ? to : [to]
+
+  return (
+    <Subscribe to={_to}>
+      {({ state }) => {
```
Comment (mzikherman) [inline src/Router/Connect.tsx]: Wondering about the signature here, if `_to` is an array of different state objects to subscribe for (let's say a component subscribes to both the global app state and a state specific to its own use), I think this callback returns two objects, so `({ state }, filterState) => ...` kinda thing.

Not sure if that's handled here (or if that's the intent to be used that way).

**Comment by mzikherman** [comment]
I'm not very opinionated here and I'm happy to go with what the consensus is, but as someone who knows very little about Redux, and esp since we've solidified React/Relay+ unstated (and even occasional direct `setState` if it makes sense), I'm not sure I'm gaining much from `Subscribe` -> `Connect`, or necessarily wanting or needing to make something more Redux-y.

**Comment by damassi** [comment]
@mzikherman yeah I’m actually inclined to agree; was a bit of late night exploring and now I don’t think it adds too much over subscribe as it just subtracts a level of nesting. Gonna close this!

**Pull request closed without merging**